commit/report.go

```diff
@@ -24,241 +23,44 @@ import (
 	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
 )
 
-// reportBuilderFunc is used to inject different algorithms for building commit reports.
-type reportBuilderFunc func(
+func encodeReports(
```

Comment:
Use split view for this function. `encodedReports` and `reportBuilderFunc` are totally different. The report encoding logic was split out from the builder and now lives in this function.

TODO: unit test for this function.